libc/src/stdio/generic/fseeko.cpp

```diff
@@ -0,0 +1,26 @@
+//===-- Implementation of fseeko ------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/stdio/fseeko.h"
+#include "src/__support/File/file.h"
+
+#include "src/errno/libc_errno.h"
+
+namespace LIBC_NAMESPACE {
+
+LLVM_LIBC_FUNCTION(int, fseeko, (::FILE * stream, off_t offset, int whence)) {
+  auto result =
+      reinterpret_cast<LIBC_NAMESPACE::File *>(stream)->seek(offset, whence);
+  if (!result.has_value()) {
+    libc_errno = result.error();
+    return -1;
+  }
+  return 0;
```

Comment:
In that case, then we can rely on the implicit conversion to `bool` for `cpp::expected` to make this cleaner:
```suggestion
  if (reinterpret_cast<LIBC_NAMESPACE::File *>(stream)->seek(offset, whence))
    return 0;
  libc_errno = result.error();
  return -1;
```

Reply:
But where are we getting the result from? Here: 

> libc_errno = result.error();

Reply:
@nickdesaulniers, I don't think the suggestion would work.

Reply:
d'oh! You're right. I guess pulling out `result` into its own statement would do the trick; at that point we're just inverting the conditional.